docs/articles/infrastructure/ci-cd/freeipa.rst

```diff
@@ -0,0 +1,240 @@
+
+.. code-block:: none
+    :caption: Example output of 'id admin'
+
+    uid=733200000(admin) gid=733200000(admins) groups=733200000(admins)
+
+.. _troubleshooting:
+
+Troubleshooting
+---------------
+
+Server
+^^^^^^
+
+If the installation fails with "Unable to restart server", look in the logs for the certificate server.
+
+.. code-block:: bash
+
+    sudo journalctl -xe -u pki-tomcatd
+
+If you find something like "Unable to stat /bin/java" it means that the java path is misconfigured. Find java through 'which java' and create a symbolic link in place of the missing binary.
+
+.. code-block:: bash
+
+    sudo ln -s /bin/java /path/to/java
+
+When an installation fails it may be necessary to delete the configuration for the certificate server created during installation. Especially if it fails with "Unable to restart server"
+
+.. code-block:: bash
+
+    # When the installation fails, delete all certificate server configuration
+    rm -rf /var/log/pki/pki-tomcat
+    rm -rf /etc/sysconfig/pki-tomcat
+    rm -rf /etc/sysconfig/pki/tomcat/pki-tomcat
+    rm -rf /var/lib/pki/pki-tomcat
+    rm -rf /etc/pki/pki-tomcat
+    rm -rf /etc/default/pki-tomcat 
+    rm -rf /etc/dogtag/tomcat/pki-tomcat
+    ipa-server-install --uninstall -U
+
+Client
+^^^^^^
+
+If you cannot find a ipa user using the id command, it's most probably because of a misconfigured nsswitch.conf Make sure to add 'sss' at the appropriate places according to the example below. The ordering is important. 'sss' should always be after 'compat' or 'files'. This dictates in which order the system will lookup users. A misconfigured nsswitch.conf may result in the inability to access the system.
+
+.. note::
+
+    "sss" refers to SSSD, a system daemon. Its primary function is to provide access to identity and authentication remote resource through a common framework that can provide caching and offline support to the system." [#SSSD]_
+
+.. code-block:: none
+
+    # /etc/nsswitch.conf
+    #
+    # Example configuration of GNU Name Service Switch functionality.
+    # If you have the `glibc-doc-reference' and `info' packages installed, try:
+    # `info libc "Name Service Switch"' for information about this file.
+
+    passwd:         compat sss
+    group:          compat sss
+    shadow:         compat sss
+    gshadow:        files
+
+    hosts:          files dns
+    networks:       files
+
+    protocols:      db files
+    services:       db files sss
+    ethers:         db files
+    rpc:            db files
+
+    netgroup:       nis sss
+    sudoers: files sss
+
+If no home directory is created upon loging in with an ipa user, use the commands below. 
+
+.. code-block:: none
+
+    # Source: https://bgstack15.wordpress.com/2017/06/26/enabling-mkhomedir-on-ubuntu-for-freeipa/
+    sed -i -r -e 's/Default:\s\w+/Default: yes/;' /usr/share/pam-configs/mkhomedir
```

Comment:
Perhaps a comment saying that it would replace whatever was written after "Default: " with "yes" would be useful here.